### PR TITLE
feat(packages/dynamodb): add type definitions

### PR DESCRIPTION
**Pull request by naorpeled**

### CODE REVIEW

**Comment by naorpeled** [comment]
@willfarrell 1/3 is done 😎

File: packages/dynamodb/index.test-d.ts [hunk 1]
```diff
@@ -1,13 +1,9 @@
-/* import middy from '@middy/core'
+import type middy from '@middy/core'
+import type { Context as LambdaContext } from 'aws-lambda'
```
Comment (naorpeled) [inline packages/dynamodb/index.test-d.ts:2]:
```suggestion
import middy from '@middy/core'
import { Context as LambdaContext } from 'aws-lambda'
```

File: packages/dynamodb/index.test-d.ts [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
 import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
 import { captureAWSv3Client } from 'aws-xray-sdk'
 import { expectType } from 'tsd'
-import dynamodb, { Context } from '.'
-
-// use with default options
-expectType<middy.MiddlewareObj<unknown, any, Error, Context<undefined>>>(
-  dynamodb()
-)
+import dynamodb, { type Context } from '.'
```
Comment (naorpeled) [inline packages/dynamodb/index.test-d.ts:6]:
```suggestion
import dynamodb, { Context } from '.'
```

File: packages/dynamodb/index.test-d.ts [hunk 3]
```diff
@@ -21,21 +17,56 @@ const options = {
...
 )
-*/
+
+// use with setToContext: true
+expectType<middy.MiddlewareObj<unknown, any, Error, LambdaContext & Record<'superSecretAccessToken', any>>>(
+  dynamodb({ ...options, setToContext: true })
+)
+
+// @ts-expect-error - fetchData is required
+dynamodb({ ...options, fetchData: undefined })
```
Comment (naorpeled) [inline packages/dynamodb/index.test-d.ts:51]:
```suggestion
```

File: packages/dynamodb/index.d.ts [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
 import { Options as MiddyOptions } from '@middy/util'
 import { Context as LambdaContext } from 'aws-lambda'
 import { DynamoDBClient, DynamoDBClientConfig } from '@aws-sdk/client-dynamodb'
 
-interface Options<AwsDynamoDBClient = DynamoDBClient>
-  extends MiddyOptions<AwsDynamoDBClient, DynamoDBClientConfig> {}
+type Options<AwsDynamoDBClient = DynamoDBClient> = Omit<MiddyOptions<AwsDynamoDBClient, DynamoDBClientConfig>, 'fetchData'>
+&
+{
+  fetchData: {
```
Comment (naorpeled) [inline packages/dynamodb/index.d.ts:9]:
```suggestion
  fetchData?: {
```

File: packages/dynamodb/index.test-d.ts [hunk 5]
```diff
@@ -21,21 +17,54 @@ const options = {
   awsClientAssumeRole: 'some-role',
   awsClientCapture: captureAWSv3Client,
   fetchData: {
-    config: {
-      Application: 'app',
-      ClientId: '0001',
-      Configuration: 'lambda-n',
-      Environment: 'development'
+    superSecretAccessToken: {
```
Comment (willfarrell) [inline packages/dynamodb/index.test-d.ts:20]: secrets should never be stored in dynamo, should use an example where a config for the lambda is passed in instead.

Comment (naorpeled) [reply packages/dynamodb/index.test-d.ts:20]: Good point 😎

**Pull request merged**